scene/3d/path_3d.cpp

```diff
@@ -303,24 +307,19 @@ void PathFollow3D::_bind_methods() {
 }
 
 void PathFollow3D::set_offset(real_t p_offset) {
-	delta_offset = p_offset - offset;
-	offset = p_offset;
-
-	if (path) {
-		if (path->get_curve().is_valid()) {
-			real_t path_length = path->get_curve()->get_baked_length();
-
-			if (loop) {
-				offset = Math::fposmod(offset, path_length);
-				if (!Math::is_zero_approx(p_offset) && Math::is_zero_approx(offset)) {
-					offset = path_length;
-				}
-			} else {
-				offset = CLAMP(offset, 0, path_length);
-			}
+	if (path && path->get_curve().is_valid()) {
+		real_t path_length = path->get_curve()->get_baked_length();
+		if (loop) {
+			p_offset = Math::fposmod(p_offset, path_length);
+		} else {
+			p_offset = CLAMP(p_offset, 0, path_length);
 		}
-
+		prev_offset = offset;
+		offset = p_offset;
 		_update_transform();
+	} else {
+		prev_offset = p_offset;
+		offset = p_offset;
```

Comment:
Why is this branch added?

Reply:
I reorganized some of the conditional branching while narrowing down what was going on, but it looks like those changes were unnecessary so I've reverted much of the changes in that function. Thank you for pointing that out :+1: